### PR TITLE
Allow Github Actions to be retriggered manually if needed

### PR DESCRIPTION
**Pull request by bwagner5**
Issue #, if available:

Description of changes:
 - Allow Github Actions to be retriggered manually if needed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ name: Release operator version, generates a new Docker image and helm charts
 on:
   push:
     tags: ['v*']
+  workflow_dispatch: {}
```
Comment (prateekgogia) [inline .github/workflows/release.yaml:5]: Lets not have the manual button for release workflow?

**Comment by prateekgogia** [comment]
Thanks I was just going to open the PR for this, saw you already have it covered :)

**Comment by ellistarn** [comment]
Any reason to not mirror Karpenter here? https://github.com/aws/karpenter/blob/main/.github/workflows/ci.yaml#L2

File: .github/workflows/release.yaml (hunk 1 above)
Comment (bwagner5) [reply .github/workflows/release.yaml:5]: I've found this helpful for re-driving a release if there's a transient failure on other projects I maintain. If you don't have the manual trigger, on an error, you'd have to remove the tag and re-release which sends notifications to watchers on the repo.

**Review by prateekgogia — APPROVED** [review]

**Comment by prateekgogia** [comment]
> Any reason to not mirror Karpenter here? https://github.com/aws/karpenter/blob/main/.github/workflows/ci.yaml#L2

I will look into mirroring this here.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (ellistarn) [reply .github/workflows/release.yaml:5]: Sounds like we need this in karpenter :)

**Comment by ellistarn** [comment]
We could look into it here: https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#access-to-reusable-workflows

**Pull request merged**